### PR TITLE
Delta with CS: core, allocator, some important fixes / improvements

### PR DESCRIPTION
**Pull request by mgonchar**

### CODE REVIEW

**Review by dmitryermilov — DISMISSED** [review]

File: _studio/shared/src/mfx_umc_alloc_wrapper.cpp [hunk 1]
```diff
@@ -337,6 +322,11 @@ UMC::Status mfx_UMC_FrameAllocator::InitMfx(UMC::FrameAllocatorParams *,
     case MFX_FOURCC_Y416:
         color_format = UMC::Y416;
         break;
+#endif
+#if defined (MFX_VA_WIN)
```
Comment (onabiull) [inline _studio/shared/src/mfx_umc_alloc_wrapper.cpp:326]: Do we need this part here?

File: _studio/shared/src/mfx_umc_alloc_wrapper.cpp [hunk 2]
```diff
@@ -1306,6 +1275,11 @@ SurfaceSource::SurfaceSource(VideoCORE* core, const mfxVideoParam& video_param,
         case MFX_FOURCC_Y416:
             color_format = UMC::Y416;
             break;
+#endif
+#if defined (MFX_VA_WIN)
```
Comment (onabiull) [inline _studio/shared/src/mfx_umc_alloc_wrapper.cpp:1279]: And here?

File: _studio/shared/src/mfx_umc_alloc_wrapper.cpp [hunk 3]
```diff
@@ -2306,13 +2280,13 @@ mfxStatus   mfx_UMC_FrameAllocator_D3D::PrepareToOutput(mfxFrameSurface1 *surfac
         {
             mfxFrameSurface1 & internalSurf = m_frameDataInternal.GetSurface(index);
             mfxFrameSurface1 surface = MakeSurface(internalSurf.Info, internalSurf.Data.MemId);
-
+            mfxU16 outMemType = static_cast<mfxU16>((m_IOPattern & MFX_IOPATTERN_OUT_SYSTEM_MEMORY ? MFX_MEMTYPE_SYSTEM_MEMORY : MFX_MEMTYPE_DXVA2_DECODER_TARGET) |
```
Comment (onabiull) [inline _studio/shared/src/mfx_umc_alloc_wrapper.cpp:2283]: DXVA2?

File: _studio/shared/src/mfx_umc_alloc_wrapper.cpp (hunk 1 above)
Comment (mgonchar) [reply _studio/shared/src/mfx_umc_alloc_wrapper.cpp:326]: yes, better to clean up

File: _studio/shared/src/mfx_umc_alloc_wrapper.cpp (hunk 2 above)
Comment (mgonchar) [reply _studio/shared/src/mfx_umc_alloc_wrapper.cpp:1279]: same, thanks

File: _studio/shared/src/mfx_umc_alloc_wrapper.cpp (hunk 3 above)
Comment (mgonchar) [reply _studio/shared/src/mfx_umc_alloc_wrapper.cpp:2283]: can change to alias ```MFX_MEMTYPE_VIDEO_MEMORY_DECODER_TARGET``` but this enum historically indicate video memory

**Review by onabiull — APPROVED** [review]

**Pull request merged**